### PR TITLE
server: use stmt should apply to subsequent stmts in multi-stmt mode

### LINKED ISSUE #26688: different behavior between mysql and tidb about use clause in transaction
### 1. Minimal reproduce step (Required)

The actual problem comes from the `syncer` tool  which is deprecated now, we have not switched to `DM` yet.

```go
package main

import (
        "database/sql"
        "log"

        _ "github.com/go-sql-driver/mysql"
)

func main() {
        db, err := sql.Open("mysql", "root@tcp(127.0.0.1:4000)/?multiStatements=true")
        if err != nil {
                log.Fatalf("open mysql err: %s", err)
        }

        txn, err := db.Begin()
        if err != nil {
                log.Fatal(err)
        }

        _, err = txn.Exec("use test; create table if not exists t1 (id int); drop table t1;")
        if err != nil {
                log.Fatal(err)
        }

        err = txn.Commit()
        if err != nil {
                log.Fatal(err)
        }

        log.Print("OK")
}
```

### 2. What did you expect to see? (Required)

MariaDB(10.3.29) produces: `OK`

### 3. What did you see instead (Required)

TiDB(v5.1.0) produces ：`Error 1046: No database selected`

### 4. What is your TiDB version? (Required)

Release Version: v5.1.0
Edition: Community
Git Commit Hash: 8acd5c88471cb7b4d4c4a8ed73b4d53d6833f13e
Git Branch: release-5.1
UTC Build Time: 2021-07-28 04:40:26
GoVersion: go1.16.5
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Issue Number: close #26688

Problem Summary:

When multi-statement is enabled, chaining statements following a "use" stmt was not successful. i.e.
```sql
use test; create table if not exists t1 (id int); drop table t1;
```

This is because of prefetching for point plans. It seems unsafe to prefetch the plan if the schema could be changed (but the query might still be valid). So my solution was to disable prefetching if there is a use stmt.

### What is changed and how it works?

What's Changed:

Fixes a bug in multi-stmt mode.

### Check List 

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [x] Performance regression: Consumes more CPU (Point plan cache will be disabled in some edge-case scenarios that are not expected to appear in most user queries).
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
When executing multiple statements in one `COM_QUERY` API call, TiDB did not correctly handle the case that a `use db` statement applies to all remaining statements.
```

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -1821,6 +1821,25 @@ func (cli *testServerClient) runTestMultiStatements(c *C) {
...
+		rows = dbt.mustQuery("SELECT IFNULL(DATABASE(),'success')")
+		if rows.Next() {
+			err = rows.Scan(&usedb)
+			c.Assert(err, IsNil)
+			c.Assert(usedb, Equals, "success")
+		} else {
+			dbt.Error("no database() result")
+		}
+		// Because no DB is selected, if the use multistmtuse is not successful, then
+		// thee create table + drop table statements will return errors.
```
Comment (xhebox) [inline server/server_test.go:1840]: then?

Comment (morgo) [reply server/server_test.go:1840]: thx, it should be the*

**Review by xhebox — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]

**Pull request merged**